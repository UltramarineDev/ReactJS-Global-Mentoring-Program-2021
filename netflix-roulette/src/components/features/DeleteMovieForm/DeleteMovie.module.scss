.header {
 font-size: 26px;
}

.content {
  padding-bottom: 36px;
}

.button {
  float: right;
  padding-right: 16px;
}