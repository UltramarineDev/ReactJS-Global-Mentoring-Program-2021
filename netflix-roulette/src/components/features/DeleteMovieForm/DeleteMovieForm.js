import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';

import { wordings } from '../../../locales/wordings';
import { buttonTypes, buttonSizes } from '../../shared/constants';
import Button from '../../shared/button/Button';
import { deleteMovieAction } from '../../../actions';

import styles from '../../../../styles/DeleteMovie.module.scss';

const DeleteMovieForm = ({ movieId }) => {
  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(deleteMovieAction(movieId));
  };

  return (
    <>
      <p className={styles.header}>{wordings.delete_movie}</p>
      <p className={styles.content}>{wordings.delete_movie_content}</p>
      <div className={styles.button}>
        <Button
          label={wordings.confirm}
          type={buttonTypes.PRIMARY}
          onClick={handleDelete}
          size={buttonSizes.SMALL}
        />
      </div>
    </>
  );
};

DeleteMovieForm.propTypes = {
  movieId: PropTypes.number.isRequired,
};

export default DeleteMovieForm;
