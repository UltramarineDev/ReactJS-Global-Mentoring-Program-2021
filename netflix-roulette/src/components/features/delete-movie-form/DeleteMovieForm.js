import React from 'react';

const DeleteMovieFormComponent = () => (
  <>
    <p>DELETE MOVIE</p>
    <p>Are you sure you want to delete this movie?</p>
  </>
);

export default DeleteMovieFormComponent;
