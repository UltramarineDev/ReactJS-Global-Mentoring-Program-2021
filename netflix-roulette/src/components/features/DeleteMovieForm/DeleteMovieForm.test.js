import React from 'react';
import { render } from '@testing-library/react';

import DeleteMovieForm from './DeleteMovieForm';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('DeleteMovie', () => {
  test('renders delete form component', () => {
    const movieId = 1;
    const { asFragment } = render(<DeleteMovieForm movieId={movieId} />);
    expect(asFragment(<DeleteMovieForm movieId={movieId} />)).toMatchSnapshot();
  });
});
