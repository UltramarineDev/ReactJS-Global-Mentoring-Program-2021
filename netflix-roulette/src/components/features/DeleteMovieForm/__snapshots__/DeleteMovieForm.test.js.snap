// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovie renders delete form component 1`] = `
<DocumentFragment>
  <p
    class="header"
  >
    DELETE MOVIE
  </p>
  <p
    class="content"
  >
    Are you sure you want to delete this movie?
  </p>
  <div
    class="button"
  >
    <button
      class="searchButton smallSize"
      type="submit"
    >
      <span
        class="labelSearch"
      >
        CONFIRM
      </span>
    </button>
  </div>
</DocumentFragment>
`;
